### PR TITLE
Don't kill container if health checks fail.

### PR DESCRIPTION
**Pull request by rculbertson**
One reason we killed the container was so the job wouldn't be stuck in STARTING
state forever, which could be confusing for user. We now have explicit
HEALTHCHECKING state, so it should be clear why the job isn't running.

### CODE REVIEW

**Comment by davidxia** [comment]
:+1:

File: helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java [hunk 1]
```diff
@@ -151,20 +149,11 @@ private int run0() throws InterruptedException, DockerException {
...
                  retryMillis, config, containerId);
-
-        totalMillis += retryMillis;
-        if (totalMillis >= HEALTHCHECK_MAX_RETRY_MILLIS) {
-          docker.killContainer(containerId);
-          throw new RuntimeException("container failed repeated health checks");
-        } else {
-          Thread.sleep(retryMillis);
-        }
+        Thread.sleep(retryMillis);
```
Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java]: Should we just keep health checking forever? The state will stay as HEALTHCHECKING. Should it be marked as FAILED?

Comment (rculbertson) [inline helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java]: If we health check forever, something at a higher level (i.e. helios-helper) will need to decide when to give up, and what action to take (probably undeploy). I think that's ok, and pretty straight forward to reason about.

But I think things get trickier if we put that logic in the agent.

* If health checks fail, should we let the container keep running? If we do, and it eventually becomes healthy, should we detect that and mark it as RUNNING? 
* Should we stop the container? If so, should helios restart it as if it exited with an error code?
* Should we reuse FAILED (which currently means your container exited with an error code) or introduce HEALTH_CHECK_FAILED?
* If we health check forever, callers can decide when to give up. If the agent gives up should we make the timeout configurable?

I'm inclined to let it health check forever, and revisit if it's proves insufficient, but maybe I'm overthinking this?

@rohansingh @matslina?

**Comment by matslina** [comment]
I'm wary of introducing new states unless we absolutely have to. It's messy enough as it is. Letting it health check forever seems fine to me. Job definition can always be extended with an optional restartIfHealthCheckFailsForMoreThanThisManySeconds or something, should we ever need that behaviour.

File: helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java (hunk 1 above)
Comment (matslina) [inline helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java]: And commenting here would of course have been more appropriate...

Comment (davidxia) [inline helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java]: Yea we can leave it for now and let helios-helper handle poll for RUNNING. If that doesn't happen within X time, we undeploy it and fail the build. We can add that logic here https://ghe.spotify.net/helios/helios-helper/blob/master/helios_helper/helios_deploy_job.py#L155

**Comment by davidxia** [comment]
:+1:

**Comment by rohansingh** [comment]
:+1:

**Pull request merged**